qiskit/algorithms/gradients/reverse_gradient/derive_circuit.py

```diff
@@ -112,13 +123,13 @@ def derive_circuit(
     # this is added as useful user-warning, since sometimes ``ParameterExpression``s are
     # passed around instead of ``Parameter``s
     if not isinstance(parameter, Parameter):
-        raise ValueError("parameter  must be  None or of type Parameter.")
+        raise ValueError(f"parameter must be None or of type Parameter, not {type(parameter)}.")
```

Comment:
Is this True that parameter can be None. The signature of the method does not imply that and if None won't this instance check fail?

Reply:
Done in [a1c7470](https://github.com/Qiskit/qiskit-terra/pull/9287/commits/a1c747025d68b12af58e71194bb03c68d110bbf9)